io/src/main/scala/sbt/io/IO.scala

```diff
@@ -454,13 +462,35 @@ object IO {
 
   /** Deletes `file`, recursively if it is a directory. */
   def delete(file: File): Unit = {
-    translate("Error deleting file " + file + ": ") {
```

Comment:
The entire delete should be wrapped in `translate`, as well as the individual walking.